src/HDF5Msg.chpl

```diff
@@ -489,26 +503,38 @@ module HDF5Msg {
                 select entryDtype {
                     when DType.Int64 {
                         var flat = toSymEntry(toGenSymEntry(entry), int);
-                        var localFlat = new lowLevelLocalizingSlice(flat.a, 0..#flat.size);
-                        writeLocalDset(file_id, dset_name, localFlat, dims, int);
+                        var localFlat: [0..#flat.size] int;
+                        ref olda = flat.a;
+                        localizeData(localFlat, olda);
```

Comment:
I'd recommend trying to use chapel's bulk communication by just doing array assignment. This would be:

```chpl
var localFlat: [0..#flat.size] int = flat.a;
```

(note that the typing is needed to avoid creating an array with the same distribution as flat)

---

Background if it's helpful:
 - `lowLevelLocalizingSlice` was designed to optimize cases where most of your slices are local slices (so you can avoid allocation) and some slices live entirely on one remote node so you can optimize and use a single communication for the slice instead of a comm per element. It was not designed to optimize this case and just falls back to grabbing an element at a time so it makes sense to me that it performed poorly. You could do something like initiate one low-level slice per locale like we do for `in1dAr2PerLocAssoc`, but I think that's overkill here and was only done in that code because we were already parallel across nodes and I didn't want to create new tasks. 
 - The "go remote and have each node aggregate elements to locale 0" in this PR should be more efficient than GETing each element, but it will create pretty high load on locale 0, and there's a GET for the local array mettadata each time. I think it'd be better to have locale 0 use a `SrcAggregator`
 - The most optimal since we're communicating contiguous indices (normally aggregators are used for random indices) would just be to have a single GET from each locale or do the coforall on and have each node do a single PUT. You could do this manually, but I'm pretty confident the array assignment will do this under the covers. Our bulk assignment has some overhead for exchanging meta-data so it won't truly be a single GET/PUT, but it should be a near trivial amount.

Reply:
Discussed offline, but forgot when I suggested this -- quick compile disable bulk transfer with `-suseBulkTransfer=false` so that needs to be unset for this to perform well. I opened https://github.com/Bears-R-Us/arkouda/issues/2311 to revisit whether it's still worthwhile to disable bulk transfer.